lib/excon/ssl_socket.rb

```diff
@@ -66,16 +66,22 @@ def initialize(data = {})
       end
 
       # maintain existing API
-      certificate_data = @data[:client_cert] || @data[:certificate_path]
-      private_key_data = @data[:client_key] || @data[:private_key_path]
-      private_key_pass = @data[:client_key_pass] || @data[:private_key_pass]
+      client_cert_data = if cert_data = @data[:client_cert_data]
+                           cert_data
+                         elsif path = (@data[:client_cert] || @data[:certificate_path])
+                           File.read path
+                         end
+
+      private_key_data = if key_data = @data[:private_key_data]
```

Comment:
Yes, thank you for being willing to discuss and work through this.

You are right that might logic shouldn't be any different in the big sense, but that the `*_path` variables are the old/deprecated ones. I like the idea of explicitly warning around them. Arguably we could remove them at some point, but we have silently accepted them for so long, that warning for a while seems like what I should have done in the first place if I wanted to actually remove it.

So, I think broadly we are on the same page for this stuff now.

Moving the logic for those to their own methods seems like a good call, it was kind of complicated already and having it inline was messy to say the least.

I don't feel particularly strongly about the instance variable version or local variable version. I guess the instance version might seem slightly better, what do you think?

Thanks!